src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/UpgradeToGeneratedRegexAnalyzerTests.cs

```diff
@@ -979,6 +979,41 @@ static void Main(string[] args)
             await VerifyCS.VerifyCodeFixAsync(test, expectedFixedCode);
         }
 
+        [Fact]
+        public async Task InterpolatedStringLiteralSyntaxFixedWhenStringLiteralIsConstantField()
+        {
+            string test = @"using System.Text.RegularExpressions;
+
+partial class Program
+{
+    const string pattern = @""a|b\s\n"";
+    const string pattern2 = $""{pattern}2"";
+
+    static void Main(string[] args)
+    {
+        Regex regex = [|new Regex(pattern2)|];
+    }
```

Comment:
What about if it's a const declared on another class, e.g. `new Regex(SomethingElse.Pattern)`?